woocommerce/wc-admin
# Add Newsletter Signup

Fixes #7450

This is a feature branch including PRs #7552, #7579, #7570, #7590, #7595.

It adds newsletter signup functionality to the first step of the profiler. This necessitates moving the "client" checkbox to the Business Details step.

A daily background process is added via the daily cron that processes the email address and marketing consent checkbox. When consented, the email address is added to the WooCommerce mailing list using an endpoint hosted at woocommere.com. There is some resiliency where if the endpoint is unavailable (e.g. due to network error), it is retried the following day.

### Screenshots

https://user-images.githubusercontent.com/9312929/131343078-d1975b0a-f6af-4632-9b47-4d0f7882bc48.mov


### Detailed test instructions:

- Start OBW and set up your browser console to monitor tracks. To do this, run `localStorage.setItem( 'debug', 'wc-admin:tracks' );`
- Observe "Get tips, product updates and inspiration straight to your mailbox" checkbox and "Email address" field in the Store Details step.
- Checking the checkbox should make the email field required, you should not be able to continue if it's not filled.
- Fill in the email address field with a valid email and click on continue.
- Observe in the track `wcadmin_storeprofiler_store_details_continue` with prop `email_signup` that appropriately flags if the user agreed to receive marketing emails.
- Continue until Business Features step.
- Observe the "I'm setting up a store for a client" checkbox in the step.
- Click on continue.
- Observe in the track `wcadmin_storeprofiler_store_business_details_continue_variant` with prop `setup_client` that appropriately flags if the user is setting up store for a client.

No changelog for this PR (multiple changelings already added in feature branch).
